drivers/platform/stm32/stm32_tdm.c

```diff
@@ -219,3 +221,54 @@ int32_t stm32_tdm_read(struct no_os_tdm_desc *desc,
 
 	return ret;
 }
+
+/**
+ * @brief Read data using SAI TDM DMA mode.
+ * @param desc - The TDM descriptor.
+ * @param data - The buffer to fill with the received data.
+ * @param nb_samples - Number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t stm32_tdm_read_dma(struct no_os_tdm_desc *desc, void *data,
```

Comment:
...continuation of my other comment:

This way, you will not implement new ops here, but rather modify existing read/write ops.
In them you will be looking of callbacks are defined and behave differently based on that.

```
if (desc->ops->tx_complete_callback)
    HAL_SAI_Receive_DMA();
else
    HAL_SAI_Receive();
```

Reply:
Got your point here.

Reply:
The callback defined check has been made with ``desc->rx_complete_callback``. Would it be relevant to add the callback and make the check with the no_os_tdm_desc?